LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/JoinRequestMessageHandler.cs

```diff
@@ -276,12 +264,13 @@ internal async Task ProcessJoinRequestAsync(LoRaRequest request)
                 var downlinkMessage = new DownlinkMessage(
                   joinAcceptBytes,
                   request.RadioMetadata.UpInfo.Xtime,
-                  loraRegion.GetDownstreamDataRate(request.RadioMetadata.DataRate, loRaDevice.ReportedRX1DROffset),
-                  loraRegion.GetDownstreamRX2DataRate(this.configuration.Rx2DataRate, null, this.logger),
-                  freq,
-                  loraRegion.GetDownstreamRX2Freq(this.configuration.Rx2Frequency, this.logger),
+                  windowToUse == Constants.ReceiveWindow2 ? default : loraRegion.GetDownstreamDataRate(request.RadioMetadata.DataRate, loRaDevice.ReportedRX1DROffset),
+                  loraRegion.GetDownstreamRX2DataRate(this.configuration.Rx2DataRate, null, logger),
+                  windowToUse == Constants.ReceiveWindow2 ? default : freq,
+                  loraRegion.GetDownstreamRX2Freq(this.configuration.Rx2Frequency, logger),
```

Comment:
Think `this.` was accidentally removed?

```suggestion
                  loraRegion.GetDownstreamRX2Freq(this.configuration.Rx2Frequency, this.logger),
```